### PR TITLE
vue/define-macros-order bug

### PR DESCRIPTION
**Pull request by edikdeisling**
Fix bugs from https://github.com/vuejs/eslint-plugin-vue/pull/1856

### CODE REVIEW

File: lib/rules/define-macros-order.js [hunk 1]
```diff
@@ -45,7 +57,11 @@ function getTargetStatementPosition(program) {
   ])
 
   for (const [index, item] of program.body.entries()) {
-    if (!skipStatements.has(item.type) && !isUseStrictStatement(item)) {
+    if (
+      inScriptSetup(scriptSetup, item) &&
```
Comment (edikdeisling) [inline lib/rules/define-macros-order.js:61]: Script setup stuff fixes multiple scripts bug(test is here: https://github.com/vuejs/eslint-plugin-vue/pull/1861/files#diff-6ebdd3c21be70e29ce94a2f399884fd220d4fe9bba358ade66b0bceb325d3102R387)

File: lib/rules/define-macros-order.js [hunk 2]
```diff
@@ -213,13 +232,12 @@ function create(context) {
     const targetComment = sourceCode.getTokenAfter(beforeTargetToken, {
       includeComments: true
     })
-    const textSpace = getTextBetweenTokens(beforeTargetToken, targetComment)
```
Comment (edikdeisling) [inline lib/rules/define-macros-order.js:216]: This fix fixes these newlines https://github.com/vuejs/eslint-plugin-vue/pull/1861/files#diff-6ebdd3c21be70e29ce94a2f399884fd220d4fe9bba358ade66b0bceb325d3102L158

File: lib/rules/define-macros-order.js [hunk 3]
```diff
@@ -228,17 +246,28 @@ function create(context) {
...
+  function getInsertText(textNode, target) {
+    const afterTargetComment = sourceCode.getTokenAfter(target, {
+      includeComments: true
+    })
+    const afterText = sourceCode.text.slice(
+      target.range[1],
+      afterTargetComment.range[0]
+    )
+    // handle case when a();b() -> b()a();
+    const invalidResult = !textNode.endsWith(';') && !/[\n;]/.test(afterText)
```
Comment (edikdeisling) [inline lib/rules/define-macros-order.js:262]: This is the third corner case I found

**Comment by edikdeisling** [comment]
I ran it in my project and the result is great. The only annoying thing left is this:
image

There is an option to introduce "newLineAfterDefineMacros: true" option maybe... For now, I've done nothing

File: lib/rules/define-macros-order.js (hunk 3 above)
Comment (FloEdelmann) [inline lib/rules/define-macros-order.js:262]: I guess it should check the start only, right?

```suggestion
    const invalidResult = !textNode.endsWith(';') && !/^[\n;]/.test(afterText)
```

File: lib/rules/define-macros-order.js (hunk 2 above)
Comment (ota-meshi) [reply lib/rules/define-macros-order.js:216]: Hmmm. I think this change is an improvement. I don't think this should be included in the patch, so could you revert this change in this PR and open another PR?

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: lib/rules/define-macros-order.js (hunk 3 above)
Comment (edikdeisling) [reply lib/rules/define-macros-order.js:262]: No, this value also get valid result
```ts
const afterText = '   ;   '
```

**Comment by edikdeisling** [comment]
@ota-meshi sure

File: lib/rules/define-macros-order.js (hunk 3 above)
Comment (FloEdelmann) [reply lib/rules/define-macros-order.js:262]: But wouldn't it also return true for `b();c()`? Or `(b();)`?

**Comment by ota-meshi** [comment]
If you write an explanation in the PR, you do not need to open an issue.

**Review by ota-meshi — APPROVED** [review]
Thank you!

File: lib/rules/define-macros-order.js (hunk 3 above)
Comment (edikdeisling) [reply lib/rules/define-macros-order.js]: The thing here is what is inside afterText. afterText contains text between the node and the next comment. So it can contain only whitespaces and new lines.
While I was writing this I understood that there is no need to look for `;`. There is no change afterText can contain it :)

Comment (edikdeisling) [reply lib/rules/define-macros-order.js]: hm

Comment (edikdeisling) [reply lib/rules/define-macros-order.js]: maybe i need to remove `;` from regexp

Comment (edikdeisling) [reply lib/rules/define-macros-order.js]: Changed it here https://github.com/vuejs/eslint-plugin-vue/pull/1863/files#diff-8293acf6fd19c2affb306c4d724be1279c64123427e207a444b5ecdab4789940R262
